src/cascade/saver/save_model_results.py

```diff
@@ -0,0 +1,92 @@
+"""
+Saves results from a model fit to a database which the EpiViz can access.
+"""
+import logging
+
+import save_results._save_results as sr
+
+from cascade.saver.generate_draws import generate_draws_table
+
+from cascade.saver import DRAWS_INPUT_FILE_PATTERN, MODEL_TITLE
+
+
+CODELOG = logging.getLogger(__name__)
+MATHLOG = logging.getLogger(__name__)
+
+
+def save_model_results(execution_context):
+    """
+    We want to visualize model results using EpiViz.  To do this, we need to
+    generate draws from a model fit, and save those draws in a database
+    that EpiViz can access.
+
+    Args:
+        execution_context (ExecutionContext): contains model id data
+
+    Returns:
+        (str) of the mvid returned by save_results
+
+    """
+
+    dm_file = execution_context.dismodfile
+
+    if not dm_file:
+        raise ValueError("Must provide a dismod file")
+
+    draws_df = generate_draws_table(dm_file)
+
+    CODELOG.debug(draws_df.columns)
+    CODELOG.debug(draws_df.iloc[0])
+
+    with execution_context.scratch_dir() as scratch_dir:
+        CODELOG.debug(scratch_dir)
+        file_path = (scratch_dir / DRAWS_INPUT_FILE_PATTERN)
+
+        model_version_id = execution_context.parameters.model_version_id
+        modelable_entity_id = execution_context.parameters.modelable_entity_id
+
+        CODELOG.debug(f"modelable_entity_id {modelable_entity_id}")
+
+        if not modelable_entity_id:
```

Comment:
Keep preconditions checks at the top of the function whenever that's practical. But this is an internal function, you shouldn't need to check this at all. This is something code at this layer should trust it's caller to get right, especially since the cost of getting it wrong is just a slightly more difficult error message a little further along.